src/VisualizationTests/HelixImageComparisonTests.cs

```diff
@@ -0,0 +1,428 @@
+﻿
+//UNCOMMENT THIS DEFINE TO UPDATE THE REFERENCE IMAGES.
+//#define UPDATEIMAGEDATA
+
+using Dynamo.Graph.Nodes.ZeroTouch;
+using Dynamo.Selection;
+using DynamoCoreWpfTests.Utility;
+using HelixToolkit.Wpf.SharpDX;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Media.Imaging;
+
+namespace WpfVisualizationTests
+{
+
+    [TestFixture]
+    public class HelixImageComparisonTests : HelixWatch3DViewModelTests
+    {
+        #region utilities
+        private static void UpdateTestData(string pathToUpdate,BitmapSource imageFileSource)
+        {
+            SaveBitMapSourceAsPNG(pathToUpdate, imageFileSource);
+        }
+
+        /// <summary>
+        /// TODO This updated function exists in Helix 2.11 - update when possible.
+        /// Renders the viewport to a bitmap.
+        /// </summary>
+        /// <param name="view">The viewport.</param>
+        /// <param name="width">The width.</param>
+        /// <param name="height">The height.</param>
+        /// <returns>A bitmap.</returns>
+        private static BitmapSource DynamoRenderBitmap(
+             Viewport3DX view, int width, int height)
+        {
+            var w = view.RenderHost.ActualWidth;
+            var h = view.RenderHost.ActualHeight;
+            view.RenderHost.Resize(width, height);
+            var rtb = view.RenderBitmap();
```

Comment:
Is it worth doing this instead of this: https://github.com/DynamoDS/Dynamo/blob/1284ce3b59df7d7f5dcd6df89f238522ae32f199/src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs#L1053 to export the background preview as an image?

Reply:
that method does not correctly resize the viewport - we want to resize the viewport *before* rendering so that we can accurately compare the pixels. Theres also no reason to export a 32bit image - we dont care about alpha channels here.

Reply:
Which is why I asked the question :)

Reply:
ahh, you mean use the new method in the save functionality? Sorry, I thought you meant the other way around.

I can try updating to helix 2.11 - then this method should be updated in both places, and we won't need to keep a custom implementation.

Let me give that a shot.

Reply:
I was wrong, this fix is in helix 2.12 that is not released yet, so we'll need to do this workaround until then - for the other function, we actually don't want to use this - there are two overloads of RenderBitmap - and that one uses the user's current window resolution so it should stay as is.